### PR TITLE
Config utils refactor and style-guide update

### PR DESCRIPTION
**Pull request by CremaLuca**
## Premise
Config doesn't need to be a class, it can be a module.
Also we said we'd update the style-guide as we go so I did.

#### refactor(`config.py`)
**~** Moved it into utils, where it belongs.
**~** It's now a module, not a class.
**+** It caches opened files for faster reads.

#### feat(`STYLE-GUIDE.md`)
**+** Added return types to methods.
**~** Fixed a couple of things I didn't like.

#### feat('README.md')
**+** Added config.json example file contents.

## Picture of a cute animal
Not an impressive PR but worth taking a look at.

Cute

### CODE REVIEW

**Comment by CremaLuca** [comment]
Linting is failing because there's still the filter example file that's all wrong.

File: test/utils/config_test.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from otri.utils import config as cfg
+import unittest
+
+class ConfigTest(unittest.TestCase):
+
+    def test_get_key_no_exception(self):
```
Comment (CremaLuca) [inline test/utils/config_test.py:6]: Codacy found an issue: [Method could be a function](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5763045)

**Comment by CremaLuca** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- test/utils/config_test.py  1
- otri/utils/config.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5763045&bid=18864764)

**Comment by CremaLuca** [comment]
## Update 28/06/2020
When using it properly I noticed that passing a default return value could be very useful, so I added it.

**Comment by CremaLuca** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- test/utils/config_test.py  1
- otri/utils/config.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5763045&bid=18864764)

File: test/utils/config_test.py (hunk 1 above)
Comment (CremaLuca) [inline test/utils/config_test.py:6]: Codacy found an issue: [Method could be a function](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5763045)

**Comment by CremaLuca** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- otri/utils/config.py  1
- test/utils/config_test.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5763045&bid=18864764)

File: test/utils/config_test.py (hunk 1 above)
Comment (CremaLuca) [inline test/utils/config_test.py:6]: Codacy found an issue: [Method could be a function](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5763045)

**Review by ricdezen — APPROVED** [review]

**Comment by CremaLuca** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- otri/utils/config.py  1
- test/utils/config_test.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5763045&bid=18864764)

File: test/utils/config_test.py (hunk 1 above)
Comment (CremaLuca) [inline test/utils/config_test.py:6]: Codacy found an issue: [Method could be a function](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5763045)

**Pull request merged**